internal/gitea/team.go

```diff
@@ -0,0 +1,149 @@
+
+// FetchOrgRepos creates a slice of RepoDetails types representing the repos
+// owned by the Gitea org.
+func FetchOrgRepos(org OrgConfig) ([]repos.RepoDetails, error) {
+	orgRepos := []repos.RepoDetails{}
+
+	gtClient, err := org.GiteaClient()
+	if err != nil {
+		return nil, fmt.Errorf("error creating gitea client: %s", err)
+	}
+
+	// Lists the gitea repositories in the org.
+	for currentPage := 1; ; {
+		opts := gitea.ListReposOptions{ListOptions: gitea.ListOptions{Page: currentPage, PageSize: giteaPerPage}}
+		gtRepos, resp, err := gtClient.ListUserRepos(org.Org, opts)
+		if err != nil {
+			return nil, fmt.Errorf("error listing repositories for gitea org: %s", org.Org)
+		}
+
+		log.Printf("Processing page %d of %d\n", currentPage, resp.LastPage)
+
+		// Iterate over repositories, populating release info for each.
+		for _, oRepo := range gtRepos {
+			r := oRepo
+			// Check if the name of the repository is in the ignore list or private or archived, or already processed.
+			if slices.Contains(org.IgnoredRepos, r.Name) || r.Private || r.Archived || repos.RepoInSlice(orgRepos, r.Name) {
+				continue
+			}
+
+			// This might actually be a monorepo. We don't have a definitive way to
+			// know that, for now, assume it is if there is a "charms" folder at the
+			// top level.
+			// TODO: Figure out some better way of determining this. Maybe it just
+			// has to be in the configuration file? If it is something like this,
+			// should we also look for a "snaps" folder as well?
+			_, _, err = gtClient.GetFile(org.Org, r.Name, r.DefaultBranch, "charms", false)
+			is_monorepo := err == nil
+
+			if is_monorepo {
+				repos := processFromMonoRepo(gtClient, org.Org, r)
+				for _, repo := range repos {
+					if len(repo.Details.Releases) > 0 {
+						orgRepos = append(orgRepos, repo.Details)
+					}
+				}
+			} else {
+				repo := processRepo(gtClient, org.Org, r)
+				if len(repo.Details.Releases) > 0 {
+					orgRepos = append(orgRepos, repo.Details)
+				}
+			}
+		}
+
+		// opendev.org gives the actual last page right up until you're getting
+		// the last page, when it suddenly becomes zero, and next page is also
+		// zero, meaning that you just loop forever. It seems like this must be
+		// a Gitea bug? Work around it.
+		if currentPage == resp.LastPage || resp.LastPage == 0 {
+			break
+		}
+		currentPage = resp.NextPage
+	}	
+
+	return orgRepos, nil
+}
+
+// Process a single (non-mono) repo.
+func processRepo(gtClient *gitea.Client, org string, oRepo *gitea.Repository) *Repository {
+	repo := &Repository{
+		Details: repos.RepoDetails{
+			Name: oRepo.Name,
+			URL:  oRepo.HTMLURL,
+		},
+		org:           org,
+		client:        gtClient,
+		defaultBranch: oRepo.DefaultBranch,
+		folder:        "",
+	}
+
+	log.Printf("processing gitea repo: %s/%s\n", repo.org, repo.Details.Name)
+
+	err := repo.Process()
+	if err != nil {
+		log.Printf("error populating repo '%s' from gitea: %s", repo.Details.Name, err.Error())
```

Comment:
You can slightly simplify here with `%s` or `%v` for errors (I usually use `%v`):

```suggestion
		log.Printf("error populating repo '%s' from gitea: %v", repo.Details.Name, err)
```

Because `fmt` will automatically call an error's `.Error()` method for you. From [fmt's docs](https://pkg.go.dev/fmt):

> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
>
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).
>
> 5. If an operand implements method String() string, that method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).